### DISCUSSION TITLE
AWS Cognitio example for Auth?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#2511] [Q&A] [answered]
We are attempting to integrate Cognitio for authorization.  I'm wondering if anyone has successfully gone down this road.

**Comment by markdav-is** [comment, marked as answer]
answering my own question:  It "just works" :)

**Reply by sbwalker** [reply]
Using OIDC I assume? I am really happy with the simplicity of the auth solution in Oqtane - it makes developers lives so much easier :)

**Reply by bfreed76** [reply]
Getting connected to Cognito using OIDC was really easy! However, when a user logs off and then tries to access the system (via Cognito's Hosted UI), Cognito automatically logs the user back in instead of asking for username and password. I believe this is expected Cognito behavior if the JWT is still valid.

In order to get around this, it seems some people have opted to save the JWT in a database on logout and then check any incoming JWTs on login against those records, and prevent login if there is a match. I'm wondering if anyone knows of a more elegant way to solve this issue.

**Reply by sbwalker** [reply]
Custom or proprietary behaviors of auth providers are difficult to accommodate... I do not have any suggestion for this scenario.

**Comment by bfreed76** [comment]
Another challenge is getting the role information from Cognito users (where it is simply stored as a custom attribute) into Oqtane. Cognito doesn't support role claims so it looks like we will need to get that info from the profile claim then assign it to the user in code. Any other ideas?

**Reply by sbwalker** [reply]
Oqtane allows you to specify the name of the claim containing the role information - it does not need to be the standard role claim.

**Reply by Ubawesome** [reply]
We're running into an issue with the OqtaneSiteAuthentiationBuilderExtension ignoring our configured RoleClaimType.
It looks like this line does a hard-coded check against the ClaimTypes.Role value:
`foreach (var claim in context.Principal.Claims.Where(item => item.Type == ClaimTypes.Role))`

We're using custom:role for the Role Claim Type, so it skips over the claim principal and fails to add users to their Cognito roles. Is there a workaround for this or something we're missing?
